staging-next 2025-01-06

vcunat opened the pull request:
##### Helpful links

- workflow docs: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#staging
- important jobs:
  https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents
  https://hydra.nixos.org/job/nixos/staging-next-small/tested
- jobsets:
  https://hydra.nixos.org/jobset/nixpkgs/staging-next
  https://hydra.nixos.org/jobset/nixos/staging-next-small
- nix-review reports: https://malob.github.io/nix-review-tools-reports/
- previous staging-next: https://github.com/NixOS/nixpkgs/pull/361878
- next staging-next: https://github.com/NixOS/nixpkgs/pull/377253
##### Important breakages
- [x] `nixosTests.containers-imperative` [below](#issuecomment-2580920875)

emilazy committed:
_7zz: remove code for macOS < 11

r-ryantm committed:
selenium-manager: 4.25.0 -> 4.27.0

r-ryantm committed:
mupdf: 1.24.11 -> 1.25.2

jtojnar committed:
gtk4: 4.16.3 → 4.16.7

https://gitlab.gnome.org/GNOME/gtk/-/compare/4.16.3...4.16.7

jtojnar committed:
pango: 1.54.0 → 1.55.5

https://gitlab.gnome.org/GNOME/pango/-/compare/1.54.0...1.55.5

jtojnar committed:
sysprof: 47.0 → 47.2

https://gitlab.gnome.org/GNOME/sysprof/-/compare/47.0...47.2

mweinelt committed:
python313Packages.watchdog: 5.0.3 -> 6.0.0

https://github.com/gorakhargosh/watchdog/blob/v6.0.0/changelog.rst

Mic92 committed:
python3.pkgs.service-identity: 24.1.0 -> 24.2.0

Changelog: https://github.com/pyca/service-identity/releases/tag/24.2.0

github-actions[bot] committed:
Merge master into staging-next

trofi committed:
SDL2: 2.30.10 -> 2.30.11

Changes: https://github.com/libsdl-org/SDL/releases/tag/release-2.30.11

ElvishJerricco committed:
systemd: add openssl as explicit buildInput (#370076)

github-actions[bot] committed:
Merge master into staging-next

Mic92 committed:
Merge remote-tracking branch 'upstream/staging-next' into HEAD

github-actions[bot] committed:
Merge staging-next into staging

Mic92 committed:
Revert "buildbot: python3Packages.service-identity: 24.1.0 -> 24.2.0"

This reverts commit 7f9857f4bdb84f223cf97029dcfe33887e2f5b19.

We update the global version now instead.

jian-lin committed:
elpa-packages: updated 2025-01-05 (from overlay)

jian-lin committed:
elpa-devel-packages: updated 2025-01-05 (from overlay)

jian-lin committed:
nongnu-packages: updated 2025-01-05 (from overlay)

jian-lin committed:
melpa-packages: updated 2025-01-05 (from overlay)

jian-lin committed:
nongnu-devel-packages: updated 2025-01-05 (from overlay)

drupol committed:
also-ucm-conf: replace hardcoded paths to `/bin/rm` and `/bin/mkdir`

drupol committed:
alsa-ucm-conf: switch to `finalAttrs` pattern

FliegendeWurst committed:
python313Packages.websockets: disable failing test, cleanup

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

chivay committed:
ghc: Always disable PIE hardening

Always disable PIE for GHC in preperation to global PIE in nixpkgs.

Currently, only musl-based stdenv enables PIE and it is explicitly
disabled by ghc derivations.

FliegendeWurst committed:
python313Packages.gevent: 24.2.1 -> 24.11.1 (#370782)

emilazy committed:
boost: boost186 -> boost187 (#369118)

mweinelt committed:
Merge remote-tracking branch 'origin/master' into staging-next

mweinelt committed:
Merge remote-tracking branch 'origin/staging-next' into staging

mweinelt committed:
python313Packages.pillow: 11.0.0 -> 11.1.0

https://pillow.readthedocs.io/en/stable/releasenotes/11.1.0.html

mweinelt committed:
python313Packages.pillow: 11.0.0 -> 11.1.0 (#371144)

mweinelt committed:
python313Packages.watchdog: 5.0.3 -> 6.0.0 (#370643)

FliegendeWurst committed:
ppp: 2.5.1 -> 2.5.2 (#370145)

nixpkgs-merge-bot[bot] committed:
byacc: 20240109 -> 20241231 (#370077)

Co-authored-by: AndersonTorres <torres.anderson.85@protonmail.com>

sternenseemann committed:
haskell.packages.ghc9121.jailbreak-cabal: 1.4 -> 1.4.1

This allows us to build packages that set `jailbreak = true` with GHC
9.12. For other versions, this update doesn't change anything, so we can
avoid the rebuild (making this change potentially cherry-pickable onto
e.g. master or staging-next). The update will otherwise happen with a
normal bump of the hackage tarball, rendering this override obsolete.

NickCao committed:
[staging] treewide: remove obsolete aligned allocation workarounds (#371054)

emilazy committed:
[staging] various: remove code for macOS < 11 (#370999)

emilazy committed:
qt5.qtModule: remove code for macOS < 11 (#371059)

FliegendeWurst committed:
ghc: Always disable PIE hardening (#371134)

emilazy committed:
SDL2: 2.30.10 -> 2.30.11 (#371067)

emilazy committed:
zlib-ng: 2.2.2 -> 2.2.3 (#369844)

emilazy committed:
bash-completion: Remove obsolete patch (#369497)

emilazy committed:
shadow: 4.16.0 -> 4.17.0 (#368930)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

emilazy committed:
gnupg24: add freepg patches (#368275)

emilazy committed:
bpftools: fix cross (#367690)

emilazy committed:
cmake: 3.30.5 -> 3.31.3 (#363395)

emilazy committed:
python312Packages.sqlparse: 0.5.1 -> 0.5.2 (#356914)

emilazy committed:
xmlto: use strictDeps (#353329)

emilazy committed:
zeromq: fix static, pkgConfigModules, move to by-name (#349742)

emilazy committed:
python311Packages.nbmake: disable parallelism (#369950)

emilazy committed:
qt6.qtwayland: backport fix for qtwayland compositor (#370520)

emilazy committed:
oniguruma: 6.9.9 -> 6.9.10 (#369960)

emilazy committed:
fontforge: fix dejavu_fonts compilation on darwin, enable Python 3.13 support (#371027)

emilazy committed:
nettle: 3.10.0 -> 3.10.1 (#369579)

emilazy committed:
python312Packages.re-assert: refactor (#369879)

emilazy committed:
kbd: 2.6.4 -> 2.7.1 (#369337)

emilazy committed:
python312Packages.pymupdf: 1.24.14 -> 1.25.1

Diff: https://github.com/pymupdf/PyMuPDF/compare/refs/tags/1.24.14...1.25.1

Changelog: https://github.com/pymupdf/PyMuPDF/releases/tag/1.25.1

emilazy committed:
cc-wrapper: Unconditionally warn about skipped native flags (#203675)

emilazy committed:
python312Packages.appdirs: fix typo (#369422)

emilazy committed:
libcamera: 0.3.2 -> 0.4.0 (#368073)

emilazy committed:
netpbm: 11.8.2 -> 11.9.0 (#369011)

emilazy committed:
python312Packages.pymupdf: set `PYMUPDF_SYSINSTALL_TEST`

emilazy committed:
wayland-protocols: 1.38 -> 1.39 (#367589)

emilazy committed:
xorg.libSM: 1.2.4 -> 1.2.5 (#367028)

emilazy committed:
xorg.libICE: 1.1.1 -> 1.1.2 (#367601)

emilazy committed:
linuxHeaders: 6.12 -> 6.12.7 (#369357)

emilazy committed:
xxHash: 0.8.2 -> 0.8.3 (#369300)

emilazy committed:
spdlog: 1.14.1 -> 1.15.0 (#354975)

emilazy committed:
glog: 0.6.0 -> 0.7.1 (#364543)

emilazy committed:
memcached: 1.6.31 -> 1.6.34 (#362219)

emilazy committed:
fluidsynth: 2.3.7 -> 2.4.2 (#358133)

emilazy committed:
python313Packages.websockets: disable failing test (#371123)

emilazy committed:
mupdf: 1.24.11 -> 1.25.2 (#369410)

emilazy committed:
flatbuffers: 24.3.25 -> 24.12.23 (#367789)

emilazy committed:
liburcu: 0.14.1 -> 0.15.0 (#366968)

emilazy committed:
libwebp: 1.4.0 -> 1.5.0 (#366783)

emilazy committed:
libtiff: switch back to CMake and fix static (#366566)

emilazy committed:
treewide: update openstack clis to python 3.12 (#363661)

emilazy committed:
freebsd: fix clang 19 build and improve overridability again (#360313)

emilazy committed:
xcbuild: fix interactive applications run by xcrun (#359721)

mweinelt committed:
python313Packages.selenium: 4.25.0 -> 4.27.1

drupol committed:
alsa-ucm-conf: fix hardcoded FHS paths (#371108)

emilazy committed:
libvdpau: fix tracing feature (#355297)

emilazy committed:
cc-wrapper: warn if clang's `--target` option is used on a wrapped compiler (#323869)

emilazy committed:
python312Packages.cffi: remove unnecessary Darwin patch

This is no longer required now that we use Apple’s `libffi`.

mweinelt committed:
selenium-manager: 4.25.0 -> 4.27.0 (#356836)

emilazy committed:
_7zz: remove code for macOS < 11 (#371281)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

rhendric committed:
python3Packages.moto: 5.0.20 -> 5.0.22 (#363883)

mweinelt committed:
python312Packages.cffi: remove unnecessary Darwin patch (#371293)

r-ryantm committed:
catch2_3: 3.7.1 -> 3.8.0

emilazy committed:
catch2_3: 3.7.1 -> 3.8.0 (#371311)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

zowoq committed:
python3.pkgs.service-identity: 24.1.0 -> 24.2.0 (#370959)

vcunat committed:
python312Packages.fs: re-add setuptools (#368857)

vcunat committed:
python3Packages.pytest-rerunfailures: 14.0 -> 15.0 (#358855)

github-actions[bot] committed:
Merge master into staging-next

r-ryantm committed:
vim: 9.1.0905 -> 9.1.0990

K900 committed:
Merge remote-tracking branch 'origin/staging-next' into staging

vcunat committed:
Merge branch 'staging' into staging-next

emilazy committed:
xorg.libXau: 1.0.11 -> 1.0.12 (#366728)

emilazy committed:
vim: 9.1.0905 -> 9.1.0990 (#371433)

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

vcunat commented:
aarch64-darwin doesn't even build stdenv so far:
https://hydra.nixos.org/build/284361443#tabs-buildsteps

emilazy committed:
cmake: remove obsolete Darwin patches

These no longer apply and shouldn’t be necessary with the new
SDK pattern.

yuyuyureka committed:
gnupg: fix cross

libgpg-error is both buildInput iand nativeBuildInput

Change-Id: Iec2f19ee770fb66aedaa444c16ebf3091b04bcef

emilazy committed:
Revert "python312Packages.cffi: remove unnecessary Darwin patch"

Fix the build on `aarch64-darwin`. Need to figure out why either
`!defined(FFI_AVAILABLE_APPLE)` or the `__builtin_available` checks
are failing, but that’s a next cycle problem.

This reverts commit eff89e3bd808ba8949356d0f4a4a61ac93cba81c.

emilazy committed:
cmake: remove obsolete Darwin patches (#371539)

emilazy committed:
Revert "python312Packages.cffi: remove unnecessary Darwin patch" (#371545)

emilazy committed:
[staging-next] gnupg: fix cross (#371542)

jian-lin committed:
emacs: elisp packages update (#371120)

emilazy committed:
folly: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebook/folly/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

emilazy committed:
fizz: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebookincubator/fizz/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

Changelog: https://github.com/facebookincubator/fizz/releases/tag/v2025.01.06.00

emilazy committed:
mvfst: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebook/mvfst/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

emilazy committed:
wangle: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebook/wangle/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

emilazy committed:
fbthrift: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebook/fbthrift/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

emilazy committed:
fb303: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebook/fb303/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

emilazy committed:
edencommon: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebookexperimental/edencommon/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

emilazy committed:
folly: add AUR patch for glog ≥ 0.7.0

emilazy committed:
fizz: add patch for glog ≥ 0.7.0

emilazy committed:
wangle: add patch for glog ≥ 0.7.0

emilazy committed:
mvfst: add patch for glog ≥ 0.7.0

emilazy committed:
fbthrift: add patch for glog ≥ 0.7.0

emilazy committed:
fb303: add patch for glog ≥ 0.7.0

emilazy committed:
edencommon: add patch for glog ≥ 0.7.0

emilazy committed:
watchman: add patch for glog ≥ 0.7.0

emilazy committed:
watchman: 2024.12.09.00 -> 2025.01.06.00

Diff: https://github.com/facebook/watchman/compare/refs/tags/v2024.12.09.00...v2025.01.06.00

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

jian-lin committed:
emacsPackages.org-link-beautify: fix build

The needed dependency qrencode is missing from Package-Requires.

adisbladis committed:
emacsPackages.org-link-beautify: fix build (#371729)

jian-lin committed:
emacsPackages.consult-gh: ignore compilation error

GaetanLepage committed:
hatch: skip failing tests

jian-lin committed:
emacsPackages.gotest-ts: fix build

github-actions[bot] committed:
Merge master into staging-next

TomaSajt committed:
rustPlatform.fetchCargoVendor: support lockfile v4 escaping

mweinelt committed:
python313Packages.poetry-core: 1.9.1 -> 2.0.0

https://github.com/python-poetry/poetry-core/blob/refs/tags/2.0.0/CHANGELOG.md

paparodeo committed:
arrow-cpp: 18.0.0 -> 18.1.0

https://github.com/apache/arrow/releases/tag/apache-arrow-18.1.0

paparodeo committed:
arrow-cpp: fix boost 187

https://github.com/apache/arrow/commit/5ec8b64668896ff06a86b6a41e700145324e1e34

mweinelt committed:
python313Packages.poetry-core: 1.9.1 -> 2.0.0 (#371828)

github-actions[bot] committed:
Merge master into staging-next

TomaSajt committed:
rustPlatform.importCargoLock: support lockfile v4 escaping

FliegendeWurst committed:
[staging-next] arrow-cpp: 18.0.0 -> 18.1.0; add boost 187 patch (#371824)

paparodeo committed:
icu: fix icu-config `/bin/sh` configure impurity

replace `/bin/sh` in the configure tests.

the icuReal derivation already does this

emilazy committed:
[staging-next] icu: fix icu-config `/bin/sh` configure impurity (#371927)

emilazy committed:
{folly,fizz,mvfst,wangle,fbthrift,fb303,edencommon,watchman}: 2024.12.09.00 -> 2025.01.06.00, add patches for glog ≥ 0.7.0 (#371610)

emilazy committed:
edencommon: skip flaky test on darwin (#369477)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

trofi committed:
shadow: 4.17.0 -> 4.17.2

Changes: https://github.com/shadow-maint/shadow/releases/tag/4.17.1
Changes: https://github.com/shadow-maint/shadow/releases/tag/4.17.2

We need this to avoid an error (unclear why exactly so far):
https://hydra.nixos.org/build/285322941/nixlog/67/tail

FliegendeWurst committed:
edencommon: skip flaky test everywhere

adisbladis committed:
emacsPackages.gotest-ts: 20241209 -> 20250107 (#371772)

JohnRTitor committed:
rustPlatform.{fetchCargoVendor,importCargoLock}: support lockfile v4 escaping (#371795)

paparodeo committed:
uhd: fix for boost187

https://github.com/EttusResearch/uhd/commit/adfe953d965e58b5931c1b1968899492c8087cf6

FliegendeWurst committed:
python312Packages.dataclasses-json: fix poetry 2 compat

FliegendeWurst committed:
python312Packages.pyrad: fix poetry 2 compat

emilazy committed:
edencommon: skip flaky test everywhere (#372055)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
libcdr: add zlib to buildInputs

> checking for zlib... no
configure: error: Package requirements (zlib) were not met:

github-actions[bot] committed:
Merge master into staging-next

MisileLab committed:
systemd: 257.1 -> 257.2 (#372292)

Signed-off-by: misilelab <misileminecord@gmail.com>

jian-lin committed:
emacsPackages.consult-gh: ignore compilation error (#371742)

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

Atemu committed:
hatch: skip failing tests (#371743)

K900 committed:
kdePackages: Frameworks 6.9 -> 6.10

K900 committed:
kdePackages.plasma-wayland-protocols: 1.15.0 -> 1.16.0

K900 committed:
kdePackages: remove unused argument

K900 committed:
[staging-next] KDE Frameworks 6.10 (#372400)

vcunat commented:
- [ ] `nix build -f pkgs/top-level/release.nix release-checks` still broken in a weird way
  https://hydra.nixos.org/build/284315748/nixlog/1/tail

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

jtojnar commented:
> `nix build -f pkgs/top-level/release.nix release-checks` still broken in a weird way
> [hydra.nixos.org/build/284315748/nixlog/1/tail](https://hydra.nixos.org/build/284315748/nixlog/1/tail)


https://github.com/NixOS/nixpkgs/blob/594dd76c494e2ea49e3bd7c9e7486a2db365a48b/pkgs/top-level/nixpkgs-basic-release-checks.nix#L83

Possibly this:

 https://github.com/NixOS/nixpkgs/blob/594dd76c494e2ea49e3bd7c9e7486a2db365a48b/pkgs/by-name/ni/nixos-build-vms/build-vms.nix#L16

jtojnar committed:
nixos-build-vm: Remove Nixpkgs path dependency

4ce241c88276a0892cc0907e805f2142b1f2b478 switched `substituteAll` for `replaceVars`.
The latter stringifies the replacements, which leads to Nixpkgs filesystem path being embedded in the output.
This is disallowed by nixpkgs-basic-release-checks.nix.
https://github.com/NixOS/nixpkgs/pull/371501#issuecomment-2581315851

mjm committed:
opencolorio: add explicit zlib input

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

wolfgangwalther committed:
nixos-build-vm: Remove Nixpkgs path dependency (#372492)

github-actions[bot] committed:
Merge master into staging-next

misuzu committed:
linux/common-config: fix typo (isLoongarch64 -> isLoongArch64)

github-actions[bot] committed:
Merge master into staging-next

ilyagr commented:
I've tried building this with my home-manager config on darwin-aarch64 (not sure why, but hope this helps), and ` fb303-2025.01.06.00` is failing.

```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.7.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/wzhdvy8pvcrfc1flrnhrbrpyb0q38pp8-source`
```

<details><summary>nix log</summary>

```
$ nix log /nix/store/a0wcrl1lipq21b5gx4znywvs8x63cv6v-fb303-2025.01.06.00.drv
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/vmcwk5vdn8ndmw6pmfy8ji12qh4x6sks-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/zcm8s5n580ivrigljdak0d769lvniwv7-glog-0.7.patch
patching file CMakeLists.txt
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/share/doc/fb303 -DCMAKE_INSTALL_INFODIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/q365b4qzms6m2i3mga6hbg9zhfq6fplj-fb303-2025.01.06.00-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/q365b4qzms6m2i3mga6hbg9zhfq6fplj-fb303-2025.01.06.00-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/x25662zlrvzy1aavv8y5vrk40bj8c8hr-clang-wrapper-19.1.6/bin/strip -DCMAKE_RANLIB=/nix/store/x25662zlrvzy1aavv8y5vrk40bj8c8hr-clang-wrapper-19.1.6/bin/ranlib -DCMAKE_AR=/nix/store/x25662zlrvzy1aavv8y5vrk40bj8c8hr-clang-wrapper-19.1.6/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00 -DBUILD_SHARED_LIBS:BOOL=TRUE -DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=TRUE -DPYTHON_EXTENSIONS:BOOL=FALSE -DINCLUDE_INSTALL_DIR:STRING=/nix/store/q365b4qzms6m2i3mga6hbg9zhfq6fplj-fb303-2025.01.06.00-dev/include -DLIB_INSTALL_DIR:STRING=/nix/store/6259hdpc107l11brwmlpd0w9rmni2qn1-fb303-2025.01.06.00/lib -DCMAKE_INSTALL_DIR:STRING=/nix/store/q365b4qzms6m2i3mga6hbg9zhfq6fplj-fb303-2025.01.06.00-dev/lib/cmake/fb303 -DCMAKE_INSTALL_PREFIX:STRING=/nix/store/q365b4qzms6m2i3mga6hbg9zhfq6fplj-fb303-2025.01.06.00-dev
[0mCMake Deprecation Warning at CMakeLists.txt:15 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

[0m
-- The CXX compiler identification is Clang 19.1.6
-- The C compiler identification is Clang 19.1.6
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/x25662zlrvzy1aavv8y5vrk40bj8c8hr-clang-wrapper-19.1.6/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/x25662zlrvzy1aavv8y5vrk40bj8c8hr-clang-wrapper-19.1.6/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- setting C++ standard to C++17
-- Found gflags from package config /nix/store/ak2wfq6d05jfw6vzppkv1dq7nzprydla-gflags-2.2.2/lib/cmake/gflags/gflags-config.cmake
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
[33mCMake Warning (dev) at /nix/store/52baknjns6l6jfd4fj1imbq9n3qr79mf-cmake-3.31.3/share/cmake-3.31/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  /nix/store/r6n6zq1fxqhx6inlw5ljyhn7dsjllb8z-folly-2025.01.06.00-dev/lib/cmake/folly/folly-config.cmake:59 (find_dependency)
  CMakeLists.txt:61 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Found Boost: /nix/store/k9wrskx9n94fss5avld0inp21i6yhn8l-boost-1.87.0-dev/lib/cmake/Boost-1.87.0/BoostConfig.cmake (found suitable version "1.87.0", minimum required is "1.51.0") found components: context filesystem program_options regex system thread
-- Found folly: /nix/store/r6n6zq1fxqhx6inlw5ljyhn7dsjllb8z-folly-2025.01.06.00-dev
-- Found ZLIB: /nix/store/h6vk8rkl8yj17s8pw8d1yh7hg04d59s3-zlib-1.3.1/lib/libz.dylib (found version "1.3.1")
-- Found folly: /nix/store/r6n6zq1fxqhx6inlw5ljyhn7dsjllb8z-folly-2025.01.06.00-dev
-- Found Sodium: /nix/store/51hijr4di10lw7c39v6dra7pvhg0q5f1-libsodium-1.0.20/lib/libsodium.dylib
-- Found folly: /nix/store/r6n6zq1fxqhx6inlw5ljyhn7dsjllb8z-folly-2025.01.06.00-dev
-- Found fizz: /nix/store/1i9hllkp7ggd7g5j5bfsmbpdlhxx8pvl-fizz-2025.01.06.00-dev
[33mCMake Warning (dev) at /nix/store/52baknjns6l6jfd4fj1imbq9n3qr79mf-cmake-3.31.3/share/cmake-3.31/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  /nix/store/ybwgav0y1h3wr2wqicsrl41d9b0flrm9-mvfst-2025.01.06.00-dev/lib/cmake/mvfst/mvfst-config.cmake:47 (find_dependency)
  /nix/store/c4d1li6bh5bnjy4hy4rhi540790jrmvz-fbthrift-2025.01.06.00/lib/cmake/fbthrift/FBThriftConfig.cmake:57 (find_package)
  CMakeLists.txt:67 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Found Boost: /nix/store/k9wrskx9n94fss5avld0inp21i6yhn8l-boost-1.87.0-dev/lib/cmake/Boost-1.87.0/BoostConfig.cmake (found version "1.87.0") found components: iostreams system thread filesystem regex context
-- Found mvfst: /nix/store/ybwgav0y1h3wr2wqicsrl41d9b0flrm9-mvfst-2025.01.06.00-dev
-- Found FBThrift: /nix/store/c4d1li6bh5bnjy4hy4rhi540790jrmvz-fbthrift-2025.01.06.00
-- Found fizz: /nix/store/1i9hllkp7ggd7g5j5bfsmbpdlhxx8pvl-fizz-2025.01.06.00-dev
-- Found wangle: /nix/store/4xkmh98qmnf2n813w7fwpwlvzdv6jh4c-wangle-2025.01.06.00
[33mCMake Warning (dev) at build/fbcode_builder/CMake/FBPythonBinary.cmake:49 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  build/fbcode_builder/CMake/FBThriftPyLibrary.cmake:4 (include)
  build/fbcode_builder/CMake/FBThriftLibrary.cmake:4 (include)
  CMakeLists.txt:71 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE) 
-- Configuring done (1.5s)
-- Generating done (0.0s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025

[0m
-- Build files have been written to: /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j10
[1/34] Generating gen-cpp2/fb303_core_constants.h, gen-cpp2/fb303_core_types.h, gen-cpp2/fb303_core_types.tcc, gen-cpp2/fb303_core_types_custom_protocol.h, gen-cpp2/fb303_core_metadata.h, gen-cpp2/BaseService.h, gen-cpp2/BaseService.tcc, gen-cpp2/BaseServiceAsyncClient.h, gen-cpp2/BaseService_custom_protocol.h, gen-cpp2/fb303_core_constants.cpp, gen-cpp2/fb303_core_data.h, gen-cpp2/fb303_core_data.cpp, gen-cpp2/fb303_core_types.cpp, gen-cpp2/fb303_core_metadata.cpp, gen-cpp2/BaseService.cpp, gen-cpp2/BaseServiceAsyncClient.cpp, gen-cpp2/BaseService_processmap_binary.cpp, gen-cpp2/BaseService_processmap_compact.cpp
FAILED: fb303/thrift/gen-cpp2/fb303_core_constants.h fb303/thrift/gen-cpp2/fb303_core_types.h fb303/thrift/gen-cpp2/fb303_core_types.tcc fb303/thrift/gen-cpp2/fb303_core_types_custom_protocol.h fb303/thrift/gen-cpp2/fb303_core_metadata.h fb303/thrift/gen-cpp2/BaseService.h fb303/thrift/gen-cpp2/BaseService.tcc fb303/thrift/gen-cpp2/BaseServiceAsyncClient.h fb303/thrift/gen-cpp2/BaseService_custom_protocol.h fb303/thrift/gen-cpp2/fb303_core_constants.cpp fb303/thrift/gen-cpp2/fb303_core_data.h fb303/thrift/gen-cpp2/fb303_core_data.cpp fb303/thrift/gen-cpp2/fb303_core_types.cpp fb303/thrift/gen-cpp2/fb303_core_metadata.cpp fb303/thrift/gen-cpp2/BaseService.cpp fb303/thrift/gen-cpp2/BaseServiceAsyncClient.cpp fb303/thrift/gen-cpp2/BaseService_processmap_binary.cpp fb303/thrift/gen-cpp2/BaseService_processmap_compact.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_constants.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_types.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_types.tcc /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_types_custom_protocol.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_metadata.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseService.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseService.tcc /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseServiceAsyncClient.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseService_custom_protocol.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_constants.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_data.h /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_data.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_types.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/fb303_core_metadata.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseService.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseServiceAsyncClient.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseService_processmap_binary.cpp /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift/gen-cpp2/BaseService_processmap_compact.cpp 
cd /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build && /nix/store/52baknjns6l6jfd4fj1imbq9n3qr79mf-cmake-3.31.3/bin/cmake -E make_directory /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift && /nix/store/c4d1li6bh5bnjy4hy4rhi540790jrmvz-fbthrift-2025.01.06.00/bin/thrift1 --legacy-strict --gen mstch_cpp2:include_prefix=fb303/thrift -I /tmp/nix-build-fb303-2025.01.06.00.drv-0/source -I /nix/store/c4d1li6bh5bnjy4hy4rhi540790jrmvz-fbthrift-2025.01.06.00/include -o /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/fb303/thrift/fb303_core.thrift
/bin/sh: line 1: 46315 Killed: 9               /nix/store/c4d1li6bh5bnjy4hy4rhi540790jrmvz-fbthrift-2025.01.06.00/bin/thrift1 --legacy-strict --gen mstch_cpp2:include_prefix=fb303/thrift -I /tmp/nix-build-fb303-2025.01.06.00.drv-0/source -I /nix/store/c4d1li6bh5bnjy4hy4rhi540790jrmvz-fbthrift-2025.01.06.00/include -o /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/build/fb303/thrift /tmp/nix-build-fb303-2025.01.06.00.drv-0/source/fb303/thrift/fb303_core.thrift
ninja: build stopped: subcommand failed.
```

</details>

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

K900 committed:
nixos/boot/loader/generic-extlinux-compatible: fix eval

K900 committed:
nixos/boot/loader/generic-extlinux-compatible: fix eval on cross as well

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
python313Packages.graphql-core: relax poetry-core pin

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

K900 committed:
kea: pin to boost186

K900 committed:
prusa-slicer: pin to boost186

K900 committed:
cabal-install: drop merged patch

mjm committed:
python313Packages.graphql-relay: relax poetry-core pin

K900 committed:
bcachefs-tools: fix build with latest liburcu

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
music-assistant: relax pillow and zeroconf constraints

github-actions[bot] committed:
Merge master into staging-next

K900 committed:
linux-hardened: condition LEGACY_VSYSCALL_NONE on x86

K900 committed:
libcap: remove runtime reference to `go` (#373067)

(cherry picked from commit 257330eed62ea5bfce1c2230f1e30fff111a8700)

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
python312Packages.pyicu: skip test for sjd locale

github-actions[bot] committed:
Merge master into staging-next

vcunat committed:
python312Packages.{dataclasses-json,pyrad}: fix poetry 2 compat (#372697)

vcunat committed:
opencolorio: add explicit zlib input (#372498)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

vcunat committed:
python3Packages.ufo2ft: 3.3.0 -> 3.4.0

https://github.com/googlefonts/ufo2ft/releases/tag/v3.3.1
https://github.com/googlefonts/ufo2ft/releases/tag/v3.4.0

The dropped patch was already included in >= 3.3.1.
This update fixes its tests; not sure what broke it in staging*:
https://hydra.nixos.org/build/285244879/nixlog/2/tail

mweinelt commented:
> I've tried building my home-manager with commit [e022056](https://github.com/NixOS/nixpkgs/commit/e0220561a9ede80dfdd4ac64f1bad63fd106e90d) config on darwin-aarch64

The build process received signal 9 (SIGKILL), so the machine probably ran out of memory.

vcunat committed:
python3Packages.pyicu: skip test for sjd locale (#373174)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/systemd-version-in-unstable-lower-than-stable/58739/6

github-actions[bot] committed:
Merge master into staging-next

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/icu-symbol-not-found-with-node-and-bun/58566/4

ilyagr commented:
> > I've tried building my home-manager with commit [e022056](https://github.com/NixOS/nixpkgs/commit/e0220561a9ede80dfdd4ac64f1bad63fd106e90d) config on darwin-aarch64
> 
> The build process received signal 9 (SIGKILL), so the machine probably ran out of memory.
> 
> Update: builds fine for me on aarch64-darwin.

Good to know, though a bit surprising. I thought 32GB are enough for everybody...

In any case, I think you're right. I also managed to build it successfully yesterday. It seems like it even builds at the same commit e022056 now.

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

vcunat committed:
linux/common-config: fix typo (isLoongarch64) (#372726)

szlend committed:
curl: apply eventfd free patch (#373692)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

uninsane committed:
libcap: use correct cross golang cross compiler (#373364)

(cherry picked from commit 619aca53daccee3bd84bfa643b48410ef1a015f1)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

vcunat committed:
rsync: 3.3.0 -> 3.4.1 (#373845)

(cherry picked from commit 0c5891d7d4529fb6af8dde020bda2145db5f873d)

jian-lin committed:
emacsPackages.consult-gh-forge: ignore compilation errors

https://hydra.nixos.org/build/285267290

jian-lin committed:
emacsPackages.consult-gh-embark: ignore compilation errors

https://hydra.nixos.org/build/285261090

vcunat committed:
rsync: fixup bad cherry-pick in 7c2121cf71fa

I'm sorry.  Bad git autoresolution.

vcunat committed:
qt6.qtdeclarative: nixfmt after 78e340633e27

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

jian-lin committed:
emacsPackages.consult-gh-{forge,embark}: ignore compilation errors (#374247)

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

dotlambda committed:
python313Packages.icalevents: 0.1.29 -> 0.2.0

Diff: https://github.com/jazzband/icalevents/compare/refs/tags/v0.1.29...v0.2.0

Changelog: https://github.com/jazzband/icalevents/releases/tag/v0.2.0

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

vcunat committed:
gnuradio*: pin back to boost186

https://hydra.nixos.org/build/285552261/nixlog/1/tail

nixpkgs-ci[bot] committed:
Merge master into staging-next

vcunat committed:
gpick: fixup build, missing #include

NickCao committed:
python313Packages.icalevents: 0.1.29 -> 0.2.0 (#374632)

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

nixpkgs-ci[bot] committed:
Merge master into staging-next

vcunat committed:
python3Packages.dbt-common: 1.12.0 -> 1.14.0

Also relax the isodate dependency.
The protobuf patch was included in the release bump.  NEWS:
https://github.com/dbt-labs/dbt-common/blob/c72ea7e3a/CHANGELOG.md#dbt-common-1140---december-4-2024

spiage commented:
I have no issues, thank you! )
